plow/driver.flow

```diff
@@ -30,12 +30,13 @@ compilePlow(cache : PlowCache, file : string) -> BModule {
 		// OK, we have an incremental module, which we read now
 		bmod = getCachedBModule(cache.modules, flowpath);
 
-		registerModuleTypes(cache, bmod);
```

Comment:
Logically, it seems correct to move this, but can you explain why?

Reply:
Because when we register types, all type dependencies must be already accounted in cache. Otherwise the transitive relation of subtyping is not correctly established.